x-pack/plugins/lens/public/indexpattern_datasource/operations/definitions/ranges/range_editor.tsx

```diff
@@ -0,0 +1,502 @@
+                />
+              )}
+            </EuiFlexItem>
+          </EuiFlexGroup>
+
+          <EuiButtonEmpty iconType="controlsHorizontal" onClick={() => onToggleEditor()}>
+            {i18n.translate('xpack.lens.indexPattern.ranges.customIntervalsToggle', {
+              defaultMessage: 'Create custom intervals',
+            })}
+          </EuiButtonEmpty>
+        </>
+      </EuiFormRow>
+    </>
+  );
+};
+
+const RangePopover = ({
+  range,
+  setRange,
+  Button,
+  isOpenByCreation,
+  setIsOpenByCreation,
+}: {
+  range: LocalRangeType;
+  setRange: (newRange: LocalRangeType) => void;
+  Button: React.FunctionComponent<{ onClick: MouseEventHandler }>;
+  isOpenByCreation: boolean;
+  setIsOpenByCreation: (open: boolean) => void;
+}) => {
+  const [isPopoverOpen, setIsPopoverOpen] = useState(false);
+  const [tempRange, setTempRange] = useState(range);
+
+  const setPopoverOpen = (isOpen: boolean) => {
+    setIsPopoverOpen(isOpen);
+    setIsOpenByCreation(isOpen);
+  };
+
+  const saveRangeIfValid = (newRange: LocalRangeType, resetRange = false) => {
+    if (resetRange) {
+      // reset the temporary range for later use
+      setTempRange(range);
+    }
+    // send the range back to the main state
+    setRange(newRange);
+  };
+  const { from, to } = tempRange;
+
+  const safeLabel = getSafeLabel(tempRange);
+
+  const gteAppendLabel = i18n.translate('xpack.lens.indexPattern.ranges.greaterThanOrEqualAppend', {
+    defaultMessage: '\u2264',
+  });
+  const gteTooltipContent = i18n.translate(
+    'xpack.lens.indexPattern.ranges.greaterThanOrEqualTooltip',
+    {
+      defaultMessage: 'Greater than or equal to',
+    }
+  );
+  const ltPrependLabel = i18n.translate('xpack.lens.indexPattern.ranges.lessThanPrepend', {
+    defaultMessage: '\u003c',
+  });
+  const ltTooltipContent = i18n.translate('xpack.lens.indexPattern.ranges.lessThanTooltip', {
+    defaultMessage: 'Less than',
+  });
+
+  return (
+    <EuiPopover
+      ownFocus
+      isOpen={isOpenByCreation || isPopoverOpen}
+      closePopover={() => {
+        setPopoverOpen(false);
+        saveRangeIfValid(tempRange, true);
+      }}
+      button={<Button onClick={() => setPopoverOpen(!isPopoverOpen)} />}
+    >
+      <EuiForm>
+        <EuiFormRow>
+          <EuiFlexGroup gutterSize="s" responsive={false} alignItems="center">
+            <EuiFlexItem>
+              <EuiFieldNumber
+                value={isValidNumber(from) ? Number(from) : ''}
+                onChange={({ target }) => {
+                  const newRange = {
+                    ...tempRange,
+                    from: target.value ? Number(target.value) : -Infinity,
+                  };
+                  setTempRange(newRange);
+                  saveRangeIfValid(newRange);
+                }}
+                append={
+                  <EuiToolTip content={gteTooltipContent}>
+                    <EuiText size="s">{gteAppendLabel}</EuiText>
+                  </EuiToolTip>
```

Comment:
This element is prepended in Visualize, not appended.

Reply:
The name it's wrong, as it should be `lte` rather than `gte`.
In the UI mockup it's an append as I suspect it maps the `N <= X < M` pattern